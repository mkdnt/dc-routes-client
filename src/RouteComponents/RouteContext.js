import React from 'react';

export default React.createContext({
    routes: [],
    addRoute: () => {},
    editRoute: () => {},
    deleteRoute: () => {},
    editing: null
})