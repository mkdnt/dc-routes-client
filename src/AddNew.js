import React, { Component } from 'react'
import RouteContext from './RouteContext'
import config from './config'

export class AddNew extends Component {
    static contextType = RouteContext

    handleClickCancel = () => {
        this.props.history.push('/')
}

    render() {
        const handleSubmit = (event) => {
            const newRoute = {
                route_name: event.target['new-route-name'].value,
                dc_area: event.target['dc_area'].value,
                distance: event.target['new-route-distance'].value,
                difficulty: event.target['difficulty'].value,
                route_type: event.target['type'].value,
                route_description: event.target['new-route-description'].value
            }
            fetch(`${config.API_ENDPOINT}/routes`, {
                method: 'POST',
                headers: {
                    'content-type': 'application/json'
                },
                body: JSON.stringify(newRoute),
            })
                .then(res => {
                    if (!res.ok)
                        return res.json().then(e => Promise.reject(e))
                        return res.json()
                })
                .then(route => {
                    this.context.addRoute(route)
                })
                .catch(error => {
                    console.error({error})
                })
                //error message here, not console
        }

        return (
            <div>
                <h2>Add New Route</h2>
        <section className='add-new-form'>
            <form onSubmit={handleSubmit}>
                <label htmlFor='new-route-name'>Course Name</label>
                <input type="text"
                id='new-route-name'
                name='new-route-name' />
                <br />
                <select className="filter" size="1">
                    <option name="dc_area" id="dc_area">DC Area:</option>
                    <option name="dc_area" id="dc_area" value="Northeast">Northeast</option>
                    <option name="dc_area" id="dc_area" value="Southeast">Southeast</option>
                    <option name="dc_area" id="dc_area" value="Northwest">Northwest</option>
                    <option name="dc_area" id="dc_area" value="Southwest">Southwest</option>
                </select>
                <br />
                <select className="filter" size="1">
                    <option name="difficulty" id="difficulty">Difficulty:</option>
                    <option name="difficulty" id="difficulty" value="Low">Low</option>
                    <option name="difficulty" id="difficulty" value="Medium">Medium</option>
                    <option name="difficulty" id="difficulty" value="High">High</option>
                </select>
                <br />
                <select className="filter" size="1">
                    <option name="type" id="type">Type:</option>
                    <option name="type" id="type" value="City Streets">City Streets</option>
                    <option name="type" id="type" value="Residential">Residential</option>
                    <option name="type" id="type" value="Trail/Path">Trail/Path</option>
                </select>
                <br />
                <label htmlFor="new-route-distance">Distance</label>
                <input type="text"
                id='new-route-distance'
                name='new-route-distance' />
                <br />
                <label htmlFor="">Description</label>
                <textarea type="text"
                id='new-route-description'
                name='new-route-description'></textarea>
                <br />
                <button className='buttons'>Submit</button>
                <button onClick={this.handleClickCancel} className='buttons'>Cancel</button>
            </form>
        </section>
            </div>
        )
    }
}

export default AddNew
