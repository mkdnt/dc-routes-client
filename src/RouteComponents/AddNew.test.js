import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer'
import {BrowserRouter} from 'react-router-dom';
import RouteContext from './RouteContext';
import AddNew from './AddNew';

describe('AddNew Component', () => {
  describe('Snapshot', () => {
    it('renders the UI as expected', () => {
      const update = renderer.create(<BrowserRouter><AddNew /></BrowserRouter>)
        expect(update).toMatchSnapshot()
    })
  })
  
  describe('Smoke test', () => {
    it('Renders without crashing', () => {
      const div = document.createElement('div');
      
      const value = {
        routes: [],
        addRoute: () => {},
        editRoute: () => {},
        deleteRoute: () => {}
        }

      ReactDOM.render(
        <BrowserRouter>
          <RouteContext.Provider value={value}>
            <AddNew />
        </RouteContext.Provider>
        </BrowserRouter>,
      div);
      ReactDOM.unmountComponentAtNode(div);
    })
  })

});