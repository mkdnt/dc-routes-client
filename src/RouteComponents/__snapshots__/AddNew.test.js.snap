// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddNew Component Snapshot renders the UI as expected 1`] = `
<div>
  <h2>
    Add New Route
  </h2>
  <section
    className="add-new-form"
  >
    <form
      onSubmit={[Function]}
    >
      <label
        htmlFor="new-route-name"
      >
        Route Name
      </label>
      <input
        id="new-route-name"
        name="new-route-name"
        onChange={[Function]}
        type="text"
      />
      <hr
        style={
          Object {
            "backgroundColor": "#011328",
            "border": " 1px solid #011328",
            "width": "75%",
          }
        }
      />
      <select
        id="dc_area"
        name="dc_area"
        onChange={[Function]}
      >
        <option
          value={null}
        >
          DC Area:
        </option>
        <option
          value="Northeast"
        >
          Northeast
        </option>
        <option
          value="Southeast"
        >
          Southeast
        </option>
        <option
          value="Northwest"
        >
          Northwest
        </option>
        <option
          value="Southwest"
        >
          Southwest
        </option>
      </select>
      <select
        id="difficulty"
        name="difficulty"
        onChange={[Function]}
      >
        <option
          value={null}
        >
          Difficulty:
        </option>
        <option
          value="Low"
        >
          Low
        </option>
        <option
          value="Medium"
        >
          Medium
        </option>
        <option
          value="High"
        >
          High
        </option>
      </select>
      <select
        id="type"
        name="type"
        onChange={[Function]}
      >
        <option
          value={null}
        >
          Type:
        </option>
        <option
          value="City Streets"
        >
          City Streets
        </option>
        <option
          value="Residential"
        >
          Residential
        </option>
        <option
          value="Trail/Path"
        >
          Trail/Path
        </option>
      </select>
      <hr
        style={
          Object {
            "backgroundColor": "#011328",
            "border": " 1px solid #011328",
            "width": "75%",
          }
        }
      />
      <label
        htmlFor="new-route-distance"
      >
        Distance (miles)
      </label>
      <input
        id="new-route-distance"
        name="new-route-distance"
        onChange={[Function]}
        style={
          Object {
            "width": "30px",
          }
        }
        type="text"
      />
      <hr
        style={
          Object {
            "backgroundColor": "#011328",
            "border": " 1px solid #011328",
            "width": "75%",
          }
        }
      />
      <label
        htmlFor=""
      >
        Description
      </label>
      <textarea
        id="new-route-description"
        name="new-route-description"
        onChange={[Function]}
        type="text"
      />
      <br />
      <button
        className="buttons"
        disabled="Route Name is required"
      >
        Submit
      </button>
      <button
        className="buttons"
        onClick={[Function]}
      >
        Cancel
      </button>
    </form>
  </section>
</div>
`;
